File: project-service/src/main/java/com/kapitonau/projectservice/service/ProjectService.java
package com.kapitonau.projectservice.service;

import com.kapitonau.projectstudio.projectservice.dto.project.ProjectPostRequest;
import com.kapitonau.projectstudio.projectservice.dto.project.ProjectResponse;

import java.util.List;

public interface ProjectService {
    List<ProjectResponse> getProjects();

    ProjectResponse getProject(Long projectId);

    ProjectResponse createProject(ProjectPostRequest body);
}

File: workspace-service/src/main/kotlin/com/kapitonau/ps/workspaceservice/service/ProjectService.kt
package com.kapitonau.ps.workspaceservice.service

import com.kapitonau.ps.apirequestlib.common.EmptyDto
import com.kapitonau.ps.apirequestlib.common.workspaces.ProjectResponse
import com.kapitonau.ps.apirequestlib.workspaces.dto.ProjectPostRequest

interface ProjectService {
    fun createProject(body: ProjectPostRequest): ProjectResponse
    fun deleteProject(projectId: Long): EmptyDto
    fun getProjectById(projectId: Long): ProjectResponse
}
